### ISSUE TITLE
OVF backward message: unguarded division yields NaN and silently corrupts B-learning

### ISSUE THREAD

**Issue by docxology** [infer-actively/pymdp#428]
## Summary

`pymdp/algos.py:234-239` (`variational_filtering_step`) performs an unguarded division that produces NaN (or inf) when the predictive next-state marginal `pred` is zero for an unreachable next state. This reverse-conditional tensor is what B (transition) learning consumes, so the NaN silently corrupts transition learning when B has structural zeros.

## Details

```python
# compute reverse conditional distribution q(z_t|z_{t+1})
cond = jtu.tree_map(
    lambda x, y, z: x * jnp.expand_dims(y, -2) / jnp.expand_dims(z, -1),
    Bs, post, pred
)
```

`pred` is the normalized prediction `jnp.sum(x * post, -1)` (`algos.py:229-231`). Whenever a B entry is structurally zero for a given source state — ubiquitous with sparse B, graph-worlds, or tile/terminal signals — the corresponding `pred` entry is exactly `0`, and the division produces `0/0 → NaN` (or `inf`).

Every other division in the module guards with `jnp.clip(..., min=MINVAL)` (see `algos.py:885, 897, 899, 1079, 1094`; `MINVAL` is imported at `algos.py:16`). This site is the lone exception.

## Suggested fix

Clip the denominator before dividing, matching the surrounding guard style:

```python
cond = jtu.tree_map(
    lambda x, y, z: x * jnp.expand_dims(y, -2) / jnp.expand_dims(jnp.clip(z, min=MINVAL), -1),
    Bs, post, pred
)
```

(`MINVAL` is already available in module scope; no new import needed.)

## Environment

- inferactively-pymdp 1.0.3 (commit 0681ce4)
- Found via static red-team review; confirmed by reading the surrounding guarded sites
